core/frontend/src/tile/RealityTileLoader.ts

```diff
@@ -143,7 +143,23 @@ export abstract class RealityTileLoader {
         reader = I3dmReader.create(streamBuffer, iModel, modelId, is3d, tile.contentRange, system, yAxisUp, tile.isLeaf, isCanceled, undefined, this.wantDeduplicatedVertices);
         break;
       case TileFormat.Gltf:
-        const props = GltfReaderProps.create(streamBuffer.nextBytes(streamBuffer.arrayBuffer.byteLength), yAxisUp);
+        /* We can obtain the base url from the tile's tree id.
+        * The id is constructed with parts from the reality data sourcekey like this - provider:format:id:itwinId
```

Comment:
Are you assuming here that the tileset was produced by mesh export service from an iModel, and will follow this specific URL scheme? Any 3d tileset that uses glTF as its payload (e.g., every 3D Tiles 1.1 tileset) is going to come through this path, with no standard URL scheme.

Are you trying to solve the general problem of needing a base URL, or the specific problem of attaching iModel tiles as a reality model (something nobody in the real world will do)?

Reply:
This is a general problem that happens when a tileset has relative URIs for textures within the .glb/.gltf. It was initially discovered trying to attach an export (the exports have relative texture URIs), but the same behavior would show up with any tileset with relative texture URIs. I'll update the PR description to make that more clear.

The main thinking was that if a tileset came down this path, and had a valid full url in its tree id, then it would need the base url to be passed along to the creation of the gltf reader. I guess this does lead to the possibility of a full valid url being present in the tree id, with full texture urls also being present in the tileset, in which case the base url wouldn't need to be passed into the creation of the reader. A better solution might be to create a separate path for only the case where relative URIs are present and leave the main path as is.